### PR TITLE
Upgrade DCL to 1.66.0

### PR DESCRIPTION
**Pull request by LucaPrete**
Supersedes https://github.com/GoogleCloudPlatform/magic-modules/pull/10531

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `security_profile_group`, `tls_inspect` to `google_compute_firewall_policy_rule`
```

```release-note:enhancement
compute: added `security_profile_group`, `tls_inspect` to `google_compute_network_firewall_policy_rule`
```

```release-note:enhancement
clouddeploy: added `custom_target` field to  `google_clouddeploy_target` resource
```

```release-note:enhancement
privateca: added `maximum_lifetime` field to  `google_privateca_certificate_template` resource
```

### CODE REVIEW

**Comment by c2thorn** [comment]
cc current DCL build gardener: @rileykarson

File: tpgtools/api/clouddeploy/samples/custom_target.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (rileykarson) [inline tpgtools/api/clouddeploy/samples/custom_target.json]: I don't immediately see what's wrong:

E0430 19:51:13.156675   24858 sample.go:202] failed to convert "custom.target.json"
F0430 19:51:13.158027   24858 sample.go:231] Error generating sample dependency custom.target.json: unexpected end of JSON input

Comment (LucaPrete) [reply tpgtools/api/clouddeploy/samples/custom_target.json]: @rileykarson me too. Given the urgency of moving to 1.66 I deleted those new files. They have nothing to do with 1.66 and they can always go in later I guess..

File: tpgtools/api/clouddeploy/samples/custom_target.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 Google LLC. All Rights Reserved.
```
Comment (rileykarson) [inline tpgtools/api/clouddeploy/samples/custom_target.yaml:1]: This needs to be deleted too

**Review by rileykarson — APPROVED** [review]
Cloud deploy tests passed in https://github.com/GoogleCloudPlatform/magic-modules/pull/10531, I'll try readding them OOB.

Firewall has no tests in DCL, but @c2thorn and @LucaPrete had discussed handling post-merge in cl/627113929. @c2thorn you'll ensure we get this tested by next release?

**Pull request merged**